src/Microsoft.Azure.WebJobs.Host/Bindings/Logger/LoggerBinding.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Loggers;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    internal class LoggerBinding : IBinding
+    {
+        private readonly ParameterInfo _parameter;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public LoggerBinding(ParameterInfo parameter, ILoggerFactory loggerFactory)
+        {
+            _parameter = parameter;
+            _loggerFactory = loggerFactory;
+        }
+
+        public bool FromAttribute => false;
+
+        public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+        {
+            if (value == null || !_parameter.ParameterType.IsAssignableFrom(value.GetType()))
+            {
+                throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Unable to convert value to {0}.", _parameter.ParameterType));
+            }
+
+            IValueProvider valueProvider = new ValueBinder(value, _parameter.ParameterType);
+            return Task.FromResult<IValueProvider>(valueProvider);
+        }
+
+        public Task<IValueProvider> BindAsync(BindingContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            ILogger logger = _loggerFactory.CreateLogger(LoggingCategories.Function);
```

Comment:
This has to be created anew on each bind, because the logger instance needs to be lifetime scoped to the function invocation?